### PR TITLE
Introduce LocalRootSpan (replacing ServerSpan)

### LINKED ISSUE #5857: Rename ServerSpan to LocalRootSpan?
I haven't thought through this but wanted to write it down as we approach instrumentation-api stability.

E.g. a common use case is to tag attributes on the local root, and LocalRoot would cover SERVER, CONSUMER, and "scheduled jobs" INTERNAL spans

### PR DESCRIPTION
**Pull request by mateuszrzeszutek**
Resolves #5857

### CODE REVIEW

File: instrumentation-api-semconv/src/main/java/io/opentelemetry/instrumentation/api/server/ServerSpan.java [hunk 1]
```diff
@@ -23,13 +26,7 @@ public final class ServerSpan {
    */
   @Nullable
   public static Span fromContextOrNull(Context context) {
```
Comment (laurit) [inline instrumentation-api-semconv/src/main/java/io/opentelemetry/instrumentation/api/server/ServerSpan.java:28]: I believe we have a couple of customers using `ServerSpan.fromContextOrNull` to find the server span. Should we attempt not to break this? Maybe instead of relying on span keys we should instrument this method when it is bundled in user code?

Comment (mateuszrzeszutek) [reply instrumentation-api-semconv/src/main/java/io/opentelemetry/instrumentation/api/server/ServerSpan.java:28]: I don't think it's a breaking change - we're not using the `SpanKey`s anymore, that's true, but the local root logic should be equivalent to that. We should be fine as long as we don't remove that class too early.

Comment (laurit) [reply instrumentation-api-semconv/src/main/java/io/opentelemetry/instrumentation/api/server/ServerSpan.java:28]: I really should have asked this in https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/5779 Let me rephrase. If user has an old version of api where
```java
public static Span fromContextOrNull(Context context) {
  return SpanKey.SERVER.fromContextOrNull(context);
}
```
in their application then would it still work after #5779 and this change? If previously span was stored under `SpanKey.KIND_SERVER` is it still stored there besides the new keys?

Comment (mateuszrzeszutek) [reply instrumentation-api-semconv/src/main/java/io/opentelemetry/instrumentation/api/server/ServerSpan.java:28]: Oh, I see, like using the old instrumentation-api with a new javaagent... no, I suppose not - the `SERVER` span key was removed after all.
That can be fixed though, by adding a new context bridge that'll connect the old `SERVER` key with the local root span.

Comment (laurit) [reply instrumentation-api-semconv/src/main/java/io/opentelemetry/instrumentation/api/server/ServerSpan.java:28]: That would we awesome.

**Review by trask — APPROVED** [review]
nice 👍

**Review by anuraaga — APPROVED** [review]
